src/bsoncxx/lib/bsoncxx/v_noabi/bsoncxx/vector.cpp

```diff
@@ -0,0 +1,119 @@
+// Copyright 2009-present MongoDB, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <bsoncxx/vector/detail.hpp>
+#include <bsoncxx/vector/formats.hpp>
+
+//
+
+#include <cstdint>
+
+#include <bsoncxx/exception/error_code.hpp>
+#include <bsoncxx/exception/exception.hpp>
+
+#include <bson/bson.h>
+
+namespace bsoncxx {
+namespace v_noabi {
+namespace vector {
+namespace detail {
+
+enum element_type : std::uint8_t {
+    signed_integer = 0,
+    unsigned_integer = 1,
+    floating_point = 2,
+};
+
+enum element_size : std::uint8_t {
```

Comment:
```suggestion
// Equivalent to bson_vector_element_size_t.
enum element_size : std::uint8_t {
```
A reference to help maintain C and C++ driver consistency.